test/jdk/java/awt/Window/WindowTitleVisibleTest/WindowTitleVisibleTestLinuxGnome.java

```diff
@@ -0,0 +1,200 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @key headful
+ * @summary Verifies the client property xawt.mwm_decor_title for Linux.
+ *          Note: the test requires GNOME Shell window manager and will automatically
+ *          pass with any other WM.
+ * @requires (os.family == "linux")
+ * @run main WindowTitleVisibleTestLinuxGnome
+ */
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.*;
+import java.nio.charset.StandardCharsets;
+import java.io.File;
+import javax.imageio.*;
+
+public class WindowTitleVisibleTestLinuxGnome
+{
+    private static WindowTitleVisibleTestLinuxGnome theTest;
+
+    private Robot robot;
+
+    private JFrame frame;
+    private JRootPane rootPane;
+
+    private Rectangle titleBarBounds;
+    private BufferedImage titleBarImageVisible;
+    private BufferedImage titleBarImageNotVisible;
+
+    private int DELAY = 1000;
+
+    public WindowTitleVisibleTestLinuxGnome() {
+        try {
+            robot = new Robot();
+        } catch (AWTException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public void performTest() {
+        constructAndShowFrame();
+
+        robot.delay(DELAY);
+
+        Insets insets = frame.getInsets();
+        Rectangle bounds = frame.getBounds();
+        titleBarBounds = new Rectangle(bounds.x, bounds.y, bounds.width, insets.top);
+        captureTitleBarVisible();
+
+        robot.delay(DELAY);
+
+        hideTitleBar();
+
+        robot.delay(DELAY);
+
+        captureTitleBarNotVisible();
+
+        if (imagesEqual(titleBarImageVisible, titleBarImageNotVisible)) {
+            throw new RuntimeException("Test failed: title bars shown and hidden are the same.");
+        }
+
+        runSwing(() -> frame.dispose());
+
+        frame = null;
+        rootPane = null;
+    }
+
+    private static boolean imagesEqual(BufferedImage img1, BufferedImage img2) {
+        for (int px = 0; px < img1.getWidth(); px++) {
+            for (int py = 0; py < img1.getHeight(); py++) {
+                int rgb1 = img1.getRGB(px, py);
+                int rgb2 = img2.getRGB(px, py);
+                if (rgb1 != rgb2) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private void captureTitleBarNotVisible() {
+        runSwing( () -> {
+            titleBarImageNotVisible = robot.createScreenCapture(titleBarBounds);
```

Comment:
`createScreenCapture` does not require to be called on EDT.

Reply:
Thanks! Moved `createScreenCapture()` out of EDT. Please, have a look at the updated code.